compiler_opt/tools/generate_default_trace_test.py

```diff
@@ -89,6 +90,13 @@ def test_api(self, mock_get_runner):
         output_performance_path=os.path.join(tmp_dir.full_path,
                                              'output_performance'),
     ):
+      with gin.unlock_config():
```

Comment:
does it need to happen here?

does it work if you move it to the setUp of the test class?

Reply:
Moving it to the `setUp` function causes gin to complain that the config is locked. This makes sense as `unlock_config` creates a context manager designed to be used with a `with` statement, so all the context will get destroyed before the actual test functions get called. I don't see any easy way to create this context in a function that covers the entire test class.

Reply:
have you tried that? gin complains here for config is locked as well, that's why there is the with gin.unlock_config() here

i think you only need the gin.parse_... in the gin.unlock_config() context for it to work. after running it, the gin overwritten is already happening, and you don't need the rest of your code to be in the context.

Reply:
Moving both the `gin.unlock_config` statement and the config parsing to the `setUp()` function works. For some reason initially I only thought to move the `gin.unlock_config()` statement to the setup which definitely did not work. Thanks for the suggestion and your persistence.